### ISSUE TITLE
clone via HTTP fails

### ISSUE THREAD

**Issue by nomas2000** [charmbracelet/soft-serve#427]
soft-serve running on AlmaLinux9 (ie. redhat)
a large repo (that I want to mirror) works via SSH, but "fails" via HTTP
... other (smaller?) repos seem to work via HTTP

Attached (log) file shows launching fresh container and then:
installing soft-serve, importing a big-repo, and 'git clone http://...' fails with 500-err
... rerun the steps to reproduce the err ...

Did not see helpful err-msgs w/ \*DEBUG\* enabled (logs below are from previous week)
```
git clone http://build-soft/linux-stable.git TRASH
Cloning into 'TRASH'...
fatal: remote-curl: bad line length character: 500
fatal: the remote end hung up unexpectedly
Nov 07 12:08:03 build-soft soft[37635]: 2023-11-07 12:08:03 DEBU request method=GET uri=/linux-stable.git/info/refs?service=git-upload-pack addr=172.x.y.z:42488
Nov 07 12:08:03 build-soft soft[37635]: 2023-11-07 12:08:03 DEBU response status="200 OK" bytes="125 B" time=11.245363ms
Nov 07 12:08:03 build-soft soft[37635]: 2023-11-07 12:08:03 DEBU request method=POST uri=/linux-stable.git/git-upload-pack addr=172.x.y.z:42488
Nov 07 12:08:03 build-soft soft[37635]: 2023-11-07 12:08:03 DEBU response status="200 OK" bytes="255 kB" time=165.572508ms
Nov 07 12:08:03 build-soft soft[37635]: 2023-11-07 12:08:03 DEBU request method=POST uri=/linux-stable.git/git-upload-pack addr=172.x.y.z:42488
Nov 07 12:08:03 build-soft soft[37635]: 2023-11-07 12:08:03 ERRO http: http: superfluous response.WriteHeader call from github.com/charmbracelet/soft-serve/pkg/web.(*logWriter).WriteHeader (logging.go:40)
Nov 07 12:08:03 build-soft soft[37635]: 2023-11-07 12:08:03 DEBU response status="500 Internal Server Error" bytes="31 B" time=10.200038ms
```
[soft-serve-alma9-container-large-repo.txt](https://github.com/charmbracelet/soft-serve/files/13343472/soft-serve-alma9-container-large-repo.txt)

**Comment by aymanbagabas** [comment]
Hi @nomas2000, thank you very much for reporting this issue. After some debugging, I found the culprit. Soft Serve HTTP server has a read timeout of 10 seconds, and it closes the connection since the repo is taking too long to read. Nevertheless, it was also handling gzip incorrectly which is the default compression `git` use with large repositories.

This now should be fixed in #428, feel free to test it out and report back 🙂

**Comment by aymanbagabas** [comment]
Released #428 in https://github.com/charmbracelet/soft-serve/releases/tag/v0.7.2
